bitcoin/src/crypto/sighash.rs

```diff
@@ -1476,9 +1476,29 @@ mod tests {
     use crate::consensus::deserialize;
     use crate::locktime::absolute;
     use crate::script::ScriptBufExt as _;
+    use crate::{Address, PublicKey, NetworkKind};
 
     extern crate serde_json;
 
+    #[test]
+    fn legacy_sighash_regression() {
+        let tx = Transaction {
+            version: transaction::Version::ONE,
+            lock_time: absolute::LockTime::ZERO,
+            input: vec![TxIn::EMPTY_COINBASE],
+            output: vec![TxOut::NULL],
+        };
+
+        let pk = "042e58afe51f9ed8ad3cc7897f634d881fdbe49a81564629ded8156bebd2ffd1af191923a2964c177f5b5923ae500fca49e99492d534aa3759d6b25a8bc971b133".parse::<PublicKey>().unwrap();
+        let addr = Address::p2pkh(pk, NetworkKind::Main);
+        let script = addr.script_pubkey();
```

Comment:
In 10af5b2fcf913d8a47f17b364c30cdc632a9ec25:

I think it'd be cleaner to just fix the bytes of the scriptpubkey rather than deriving it like this.